### PR TITLE
Famous Artists API 

### PR DESCRIPTION
**Pull request by Izzibizz**
## Onrender link
https://izabels-first-api.onrender.com

Frontend (under construction still)
##Github repo link
https://github.com/Izzibizz/artsome

##Netlify Link
https://artsome.netlify.app/

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,30 +1,79 @@
...
+  if (showAlive) {
+    artistsOfGender = artistsOfGender.filter((artist) => artist.alive === true)
+  }
+  if (showDeceased) {
+    artistsOfGender = artistsOfGender.filter((artist) => artist.alive === false)
+  }
+  if (artistsOfGender.length > 0) {
+    res.json(artistsOfGender)
+  } else {
+    res.status(404).json({ error: 'Could not find artists of this kind, try endpoints: female / male. You can also add alive=true or alive=false at the end' })
```
Comment (maikanetaka) [inline server.js:55]: Good that not only catching the error but also very user-friendly message!

File: README.md [hunk 2]
```diff
@@ -1,13 +1,28 @@
...
 
 ## View it live
 
-Every project should be deployed somewhere. Be sure to include the link to the deployed project so that the viewer can click around and see what it's all about.
+https://izabels-first-api.onrender.com
+
+Endpoints are:  /artists
+                /artists/:name
+                /gender/:gender
+                /gender/:gender?alive=true / false
```
Comment (maikanetaka) [inline README.md:23]: The way you show the gender route that handles query parameters for "alive" and deceased is thoughtful design! It’s a nice touch that enhances the functionality beyond just filtering by gender only 👏

**Review by maikanetaka — COMMENTED** [review]
Your work, especially adding extra features and coding for the frontend, is really inspiring!🌟 I can tell you enjoy coding and your designs are super as usual! I love from the start, loading animation 😊
When all projects are done and start review/improve my code, I will check yours for inspiration. Great jobbb! 👏

File: server.js [hunk 3]
```diff
@@ -1,30 +1,79 @@
...
+app.use(express.json())
 
 // Start defining your routes here
-app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
-});
+app.get('/', (req, res) => {
+  const endpoints = expressListEndpoints(app)
+  res.json(endpoints)
+})
```
Comment (AntonellaMorittu) [inline server.js:18]: would be good to add to the documentation how to filter data using the query `alive` like:

```javascript
app.get('/', (req, res) => {
  const endpoints = expressListEndpoints(app).map(endpoint => {
    // Add alive query parameter to the '/artists/:name' endpoint
    if (endpoint.path === '/artists/:name') {
      endpoint.query = [{
        parameter: "alive",
        description: "Filter artists by whether they are alive or deceased.",
        type: "boolean",
        example: "true"
      }];
    }
    return endpoint;
  });

  res.json(endpoints);
});
``` 

in this way we inform other devs about how to fully use the api! ⭐

**Review by AntonellaMorittu — APPROVED** [review]
Hi Izabel, congrats for completing your first API, very impressive you develop the front-end for it, super! Your app meets the requirements and effectively retrieves information about artists, including filtering by name, gender, and artistic technique. I left a comment about including the query into the documentation, take a look - anyhow you passed 👏

**Pull request closed without merging**